### ISSUE TITLE
WSL2 distros fail on start with Error 0xffffffff. (Exit code 4294967295 if launched from Windows Terminal)

### ISSUE THREAD

**Issue by trbenning** [microsoft/WSL#4929]
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
**Microsoft Windows [Version 10.0.19041.84]**

* What you're doing and what's happening:  (Copy&paste the full set of _specific_ command-line steps necessary to reproduce the behavior, and their output. Include screen shots if that helps demonstrate the problem.)
**Problem occurrs anytime I try to launch a WSL2 distro.**

* What's wrong / what should be happening instead:
**All WSL2 distros crash on start. WSL1 distros work fine.
My current set:**
```powershell
C:\Users\10057871> wsl -l -v
  NAME                   STATE           VERSION
* Ubuntu                 Stopped         2
  Alpine                 Stopped         1
  docker-desktop-data    Stopped         2
  docker-desktop         Stopped         2
```

* Strace of the failing command, if applicable:  (If `some_command` is failing, then run `strace -o some_command.strace -f some_command some_args`, and link the contents of `some_command.strace` in a [gist](https://gist.github.com/) here).
**N/A**

* For WSL launch issues, please [collect detailed logs](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs).
[wsl_logs.zip](https://github.com/microsoft/WSL/files/4252638/wsl_logs.zip)

* Additional info:

I recently got a new laptop. Before configuring WSL or Docker, I upgraded to insiders version 2004 (full version above). I then enabled WSL, installed Ubuntu, & set the version to 2. Later I installed Docker for Windows, and configure that to use WSL2 as well.

Everything worked great for a while, until my group policy forced me to enable Bitlocker. After that, I started seeing the errors mentioned above. While searching for a solution, I came across [this reply on another issue](https://github.com/microsoft/WSL/issues/4103#issuecomment-501511504). After executing those 2 commands and rebooting, I was able to install Ubuntu again & upgrade it to WSL2. I then reinstalled Docker, and configured it for WSL2 again as well. This worked great for almost 2 days, until my laptop Green Screened and rebooted (I have the minidump if you'd like me to send it). After the reboot, my WSL2 instances no longer work, and I'm unable to get them working again. I even executed the `fsutil` commands again just in case they got reset, but to no avail.

Is this behavior caused by Bitlocker, or is the timing merely coincidental?

**Comment by privacyguy123** [comment]
I'm currently experiencing the same issue, minus Docker & Bitlocker. My Debian was working perfectly in WSL 2 up until today - now it hangs which causes Windows Terminal to hang also, I've tried remove/readding features etc to no avail.

**Comment by privacyguy123** [comment]
Update: today when I booted into Windows WSL said it had no distro installed despite having Debian installed up until until last night.

**Comment by slecorvaisier** [comment]
Same issue here with my WSL2 distro (ubuntu 18.04)

**Comment by Biswa96** [comment]
Do you have latest Windows 10 insider build?

**Comment by privacyguy123** [comment]
> 
> 
> Do you have latest Windows 10 insider build?

Who would put themselves through that? I'm on Slow ring atm.

**Comment by trbenning** [comment]
I'm on the slow ring as well. Interestingly enough, today I was able to create a new instance of Alpine, and upgrade it to WSL2 (I couldn't do this yesterday). Nothing about my system has changed that I'm aware of, aside from an additional reboot.

That gave me the confidence to enable WSL2 on my Docker install, which worked as well.

I just got my Ubuntu dev environment setup again, so I'm hesitant to upgrade that one, but I'll update this thread if my Alpine or Docker instances start crashing again.

**Comment by privacyguy123** [comment]
I think the most puzzling part of all this is that there are no steps to reproduce in my case - it just happened by itself overnight.

**Comment by trbenning** [comment]
My experience was more or less the same. I work primarily out of the WSL Ubuntu instance, and everything was going great until suddenly it wasn't. No idea what caused the green screen, but after rebooting I couldn't get anything to work with WSL2. I rebooted a couple of times, and reinstalled the WSL distros but was unable to upgrade them to v2. Then today, on a whim, I tried again and everything just worked. 🤷‍♂️

**Comment by privacyguy123** [comment]
This post fixed it for me!

https://www.surfacetablethelp.com/2018/10/fix-hyper-v-not-working-after-windows-10-v1809-upgraded.html

Still couldn't tell you how those options got set in the first place.

**Comment by slecorvaisier** [comment]
The post shared by @privacyguy123 solved my issue too, it just requires an additional reboot.

**Comment by pnunn** [comment]
I'm really bummed out by this.. I needed to use traceroute and couldn't under version 1 so followed the bouncing ball to upgarade to V2 and my ubuntu machine took ages to migrate. 

Now nothing works. 

I tried the post shared by @privacygui123 as well, but when I try and change the settings for vmcompute I get an error

`Unexpected error. Sorry, we ran into a problem. Please try again.`

But, try as I might... I can't get anything working.

ver returns Version 10.0.19041.113

wsl -l -v returns
Ubuntu-18.04  Stopped  2

Any ideas please.. I'm stuck without wsl working.

**Comment by Biswa96** [comment]
Try to install Ubuntu in WSL2 from scratch without migrating from v1.  `wsl.exe  --set-default-version 2` will set WSL2 as default. Also make a backup of existing WSL user home folder and installed packages.

**Comment by pnunn** [comment]
Thanks Biswa96, how do I backup the home and installed packages? Sorry, new to wsl.

**Comment by pnunn** [comment]
I tried install openSuse-Leap-15-1 after setting the default version but it won't work either.

 `wsl --distribution openSuse-Leap-15-1 --user root
The attempted operation is not supported for the type of object referenced`.

Running it as a normal user, started, set up the new user, then crashed with a screen full of the same messages.

**Comment by trbenning** [comment]
It happened again today. None of my WSL2 instances will start due to the same error. This time, I believe it was caused by me messing with Hyper-V. I needed to reserve some ports that Hyper-V was hogging, so I performed the following steps:

1. Opened Windows Features dialog, and disabled Hyper-V
2. Restarted
3. Opened PowerShell, and ran the following:
```powershell
netsh int ipv4 add excludedportrange tcp 8000 100 persistent
netsh int ipv4 add excludedportrange tcp 8800 100 persistent
```
4. Opened Windows Features dialog, and enabled Hyper-V
5. Restarted
6. After booting, all of my WSL2 instances crash with the above-mentioned error.

**Comment by pnunn** [comment]
Is there a fix for this or do I blow windows away and install a proper OS? This is crazy.

**Comment by privacyguy123** [comment]
Still no official response for this?

Are you guys on Slow or Fast ring, can we confirm the problem exists in both?

**Comment by pnunn** [comment]
I'm on slow.

**Comment by trbenning** [comment]
> Is there a fix for this or do I blow windows away and install a proper OS? This is crazy.
> 
> OK, I've converted my main machine back to v1 so I can atleast do some work.

@pnunn If this is your attitude, maybe you shouldn't be on the insider builds at all. The point of us using these builds is to find issues like these so they can fix them before the GA release.

**Comment by craigloewen-msft** [comment]
The error 0xffffffff indicates that the virtual network could not be created. We've identified a possible bug fix for this that is on Windows version 19555 and higher. 

For an immediate fix for this error, please consider moving to the fast ring (and understand that this means you'll get updates quicker but could experience more technical issues) otherwise eventually a build will be released to the slow ring that includes this fix!

As well, please let us know if the update does fix your error, we want to make sure that the fix targets this specific use case since the fix was lower in the stack.

For more context check out this Github issue where we are tracking the progress: https://github.com/microsoft/WSL/issues/4364

Thanks for helping report this!

**Comment by pnunn** [comment]
Thanks @craigloewen-msft atleast we know what it is now... I don't think I can risk the fast ring breaking more than already is. Will we know when this fix is pushed down to the slow ring? Can you let us know here perhaps so we can try wsl2 again?

Ta
Peter.

**Comment by craigloewen-msft** [comment]
Yup! I will ping here or on the linked thread when this fix becomes available on the slow ring.

**Comment by pnunn** [comment]
Any news on this one @craigloewen-msft? Still chugging away on WSL1 at this point.

**Comment by craigloewen-msft** [comment]
It's not in slow ring yet!

**Comment by daviholandas** [comment]
for me was solved like this:
1 - Uninstall docker;
2 - Uninstall / install Hyper-v;
3 - Uninstall / install subsystem;

started working again!

**Comment by pnunn** [comment]
That didn't work for me. I tired installing a new Debian machine and it won't accept the username that it asks for during startup.

The attempted operation is not supported for the type of object referenced.

is the error.

**Comment by pnunn** [comment]
Why are Microsoft putting out blog posts like this
https://devblogs.microsoft.com/commandline/wsl2-will-be-generally-available-in-windows-10-version-2004/
when wsl2 is not working?

**Comment by pnunn** [comment]
Hello.. is any one home?

**Comment by LFBernardo** [comment]
The latest version of Docker desktop decimates WSL. The only way to fix it is to remove any WSL apps and remove the entire Windows subsystem for linux. Reboot and re-install everything. I am on the latest version of everything.

**Comment by pnunn** [comment]
Unfortunately I've never installed Docker on here because I want to use Virtual Box. Is that a problem perhaps? Hmm...

**Comment by pnunn** [comment]
Well.. that went well. Just restarted for the latest update and crashed the entire laptop. No disk found.. thank goodness they rolled back. Updates now turned off for as long as possible...

What the hell Microsoft??

**Comment by pnunn** [comment]
OK, back on the substantive issue. Just removed virtual box and rebooted, wsl2 still crashes horribly.

**Comment by pnunn** [comment]
Is this ever going to get fixed. I see the media reports seem to suggest WSL2 is ready to go.. I beg to differ.

**Comment by trbenning** [comment]
For what it's worth, I haven't had any issues since the last time I posted.

**Comment by mchubby** [comment]
I did as suggested:

1. uninstalled docker, hyper-v
1. uninstalled WSL2 MSI kernel update
1. uninstalled windows capabilities WSL and virtual machine platform
1. rebooted
1. installed WSL and virtual machine platform
1. rebooted
1. installed WSL2 MSI kernel update

At this point, distros and WSL2 worked fine. **No** svchost listening on UDP 53.

1. installed hyper-v and sub-features
1. rebooted

WSL2 returns 0xffffffff, svchost **does** listen on UDP 53.

So clearly for me, there's something going on with Hyper-V on Win10 Pro

**Comment by prom3theu5** [comment]
Definitely still an issue.

Atm, I quit Docker for Desktop when it fails to start the WSL2 backend, and run the following in an Admin Powershell.

```
$processes = Get-Process -Id (Get-NetUDPEndpoint -LocalPort 53).OwningProcess | Select-Object -ExpandProperty Id

foreach ($process in $processes) {
    Stop-Process -ID $process -Force
}
```

There are always two processes it seems - DockerD, and an unknown service captured by svchost
I have no idea what the ramifications of me simply killing the tasks like I do are, but Docker for Windows starts up fine afterwards, Starts the WSL2 backend successfully - and my system is stable

**Comment by windowsair** [comment]
19608. issues still occur frequently...

**Comment by onomatopellan** [comment]
@prom3theu5 Once you get the PId with Get-Process you can open Task manager -> Services tab and sort by PId. In my case service SharedAccess (Internet Connection Sharing) uses port 53.

**Comment by roy-bentley** [comment]
After thinking about the PowerShell snippet @prom3theu5 posted [above](https://github.com/microsoft/WSL/issues/4929#issuecomment-616383430), I've worked around this on my machine. Rebooted to test a couple times, worked every time so far.

- Docker and Docker Desktop services should depend on LxssManager
- LxssManager should depend on Internet Connection Sharing

LxssManager is a system service, which means `sc` gives a permissions error when you try to config it, so adding the dependencies has to be done in the registry. I needed to do it more than once, so I made [a thing](https://gist.github.com/roy-bentley/882dc704fffb2a2921197829336fda41), which may or may not cause your PC to implode.

**Comment by Eric2XU** [comment]
Its port 53 for me due to Cisco AnyConnect VPN that installed its umbrella client. Anyone understand why Port 53 has to be open for WSL/Docker/Hyper-V? Disabling the Cisco AnyConnect Services temporarily then restarting the Hyper-V Compute Service, then my WSL2 and Docker fired right up. I then turned on the cisco services. I have a feeling I will have to do each reboot :(

**Comment by Eric2XU** [comment]
Just as a final post-reboot thought... setting Cisco to manual helps as it wont take over however it does mean I manually have to turn it back on after rebooting but that is preferable. 

Also to those not facing the cisco issue, I continued to have other issues but in general simply restarting that Hyper-V Compute Service then trying to fire up WSL2 worked most times.

**Comment by rwasef1830** [comment]
@roy-bentley workaround didn't work for me, in addition I only have Docker Desktop service (I don't have the other docker service).

**Comment by mrpond** [comment]
I fix it by doing this,
Goto device manager and remove all hyper v network adapter.. reboot and it working fine.
![image](https://user-images.githubusercontent.com/18654771/83344415-e4c38d00-a330-11ea-8013-65fdcc2c498f.png)
I notice after running wsl and got Error: 0xffffffff because it fail to create a wsl network adapter
like as @craigloewen-msft said.
![image](https://user-images.githubusercontent.com/18654771/83344450-4e439b80-a331-11ea-8cc2-05ca489b13fc.png)

**Comment by muyufan** [comment]
for me, simply un-install Hyper-V and re-start computer, then everything works

**Comment by nckdhl** [comment]
Uninstalling Hyper-V according to these instructions:
[answers.microsoft.com/en-us/windows/forum/windows_8-windows_install/how-do-i-uninstall-hyper-v](https://answers.microsoft.com/en-us/windows/forum/windows_8-windows_install/how-do-i-uninstall-hyper-v/7d268911-47cd-4c52-bfe5-ea41e58067ab )
and then restarting also solved the problem for me.

**Comment by jtsalten** [comment]
It happen to me also. I've been working with WSL2 since late May... it has been running during some days without touching. Today, when I tried to enter again in WSL I got that error too... no way to "restart" this... it just says:

PS C:\..\system32> wsl
Error: 0xffffffff

It's unbelievable... such unestability... a pity, because the idea is great. But if you cannot trust on it, it's useless... let's hope I didn't lose everything when I manage to get it back to work.

**Comment by jtsalten** [comment]
uninstalling Hyper-V ans restarting DIDN?T work.... aaagggghhhh.... what the hell!??!? With this kind of errors you really don't know what to do!!!

**Comment by el-schneider** [comment]
deactivating Acrylic DNS Proxy fixed it for me

**Comment by pit-hub** [comment]
> deactivating Acrylic DNS Proxy fixed it for me

I'm using dnscrypt-proxy. So in my case stopped the dnscrypt-proxy tried the wsl command twice and my default WSL instance booted, docker started, and I started dnscrypt-proxy. Everything is running! It seams that there is DNS resolution issues when running local DNS resolver!
Some transparency how it works will definitely help!

**Comment by somanythings** [comment]
> Its port 53 for me due to Cisco AnyConnect VPN that installed its umbrella client. Anyone understand why Port 53 has to be open for WSL/Docker/Hyper-V? Disabling the Cisco AnyConnect Services temporarily then restarting the Hyper-V Compute Service, then my WSL2 and Docker fired right up. I then turned on the cisco services. I have a feeling I will have to do each reboot :(

Thanks Eric, same problem here, but with NordVpn. Only needs to be disconnected, and both wsl2 and wsl2 docker-desktop will start.

**Comment by zanesand** [comment]
1000% this for me - Its port 53 for me due to Cisco AnyConnect VPN that installed its umbrella client.

**Comment by dcharlespyle** [comment]
Now happening to me after updating to build 20175.  Except, I am not using WSL2.
![image](https://user-images.githubusercontent.com/5110398/88311517-33a4f780-ccce-11ea-96b3-409f30949858.png)
![image](https://user-images.githubusercontent.com/5110398/88311585-4f100280-ccce-11ea-9124-b429239f0b07.png)
Now both WSL 1 and 2 are broken since upgrading to the current build.  So far, nothing has worked to fix it.

**Comment by Rebely22** [comment]
I disable Hyper-v virtual Ethernet(VEthernet) in control panel,  then start wsl2.
That error disappear and wsl start successfully.
then I enable virtual Ethernet, everything works fine for now.

**Comment by dcharlespyle** [comment]
Sadly, that didn't work for me.  It's still broken.  But I was hoping it would make a difference.

**Comment by onomatopellan** [comment]
@dcharlespyle Your issue in WSL1 is #5633. Try shortening the Windows %PATH%.

**Comment by dcharlespyle** [comment]
After cutting down the %PATH% variable to 989 characters (including spaces), WSL starts working again.  Unfortunately, this is on a machine with developer environments and Windows Kits.  Cutting down the %PATH% permanently would be problematic.  Hope they fix this problem with WSL soon.

**Comment by LewisPringle** [comment]
> Still no official response for this?
> 
> Are you guys on Slow or Fast ring, can we confirm the problem exists in both?

I have the same problem. Sometimes rebooting fixes it. Sometimes it just starts again. And I'm on 'no ring' - WinVer 2004.

**Comment by anil3a** [comment]
I had this issue when i was updating my windows to new build, I had a popup of "Windows build going to expire" every time i open my computer so I research a lot for this solution and came to know that I have switch the Insider setting from Beta or Slow ring to Dev channel and update/upgrade windows build. Once it updated I switch back to Beta channel now and this expire popup went away.
Now, with this I had new issue, that is when i open my WSL Ubuntu terminal, it was blank. If I press anything, I received I/O operation error all the time. So, now I came to this Issue here and read all of them. I looks like i have re-enable Hyper virtual...., or WSL path ...., it was confusing for me and i know it worked for some people here. 

For my case, ( I am not sure if everyone had similar situation to mine ). what i did to solve this is:
1. `dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart (enable Feature WSL)`
2. ` dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart  ( enable Feature Virtual Machine )`

and ran this
` wsl --set-default-version 2`
Set default version to 2 and see if i can set to new WSL version.

`wsl -l -v`
Check what version of Ubuntu do i have

`wsl --set-version Ubuntu-18.04 2`
Set version of Ubuntu to 2. 

Then, I got this message:
```
Conversion in progress, this may take a few minutes...
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
WSL 2 requires an update to its kernel component. For information please visit https://aka.ms/wsl2kernel
```

So, I went to that link and downloaded Kernel `wsl_update_x64.msi` file and installed, all went good.

After installation, I ran this command again,
`wsl --set-version Ubuntu-18.04 2`

Voila, done, I opened my WSL ubuntu terminal, and its all working like before.

**Comment by ProFL** [comment]
I went through the same issue and it comes out it was caused by [Acrylic DNS Proxy](https://mayakron.altervista.org/support/acrylic/Home.htm).

Maybe services like that mess with networking settings necessary for Hyper-V to work (I noticed not even my plain Hyper-V VMs would come up).

I "solved" it by uninstalling Acrylic altogether and resorting to plain old hosts file editing.

**Comment by robdunbar** [comment]
Established a VPN using Pulse Secure before starting WSL was my issue. Thanks for everyone notes.
1. Restart PC (Just disconnecting from VPN doesn't work)
2. Start WSL
3. Start VPN
VPN and WSL now work in "harmony"

**Comment by huskerjeff** [comment]
Really too bad wsl will not work consistently - I just get 
❯ wsl
Error: 0xffffffff

none of the above fixes worked for me.

**Comment by dcharlespyle** [comment]
> Really too bad wsl will not work consistently - I just get
> ❯ wsl
> Error: 0xffffffff
> 
> none of the above fixes worked for me.

Open up Task Manager and click the Performance tab.  Do you have an "Ethernet (WSL)" adapter in the list?
![image](https://user-images.githubusercontent.com/5110398/93114956-8076d000-f678-11ea-9aa3-73fc36cba3d6.png)
At the far right it gives the name of the adapter in Device Manager.  Open up Device Manager, and then try disabling it, and reenabling it.  It might be hidden.  If it is, show the hidden devices by clicking **View**, and then **Show hidden devices**.  If you don't see it in Device manager, make sure to shut down WSL.  If that doesn't work, try deleting the device and rebooting.

**Comment by QuAzI** [comment]
The same issue after upgrade to Win10 b20.04 and fixing issues with docker.
`docker network prune` drops WSL? Why? And how to fix it?
![image](https://user-images.githubusercontent.com/1897605/93991175-a85ee580-fd94-11ea-9652-01b3e309aa99.png)

**Comment by aleksandrshvets** [comment]
netsh winsock reset + restart fixes this issue for me

**Comment by QuAzI** [comment]
After `docker network prune` run Hyper-V Manager -> Hyper-V Switch Manager -> Virtual Switches
Create WSL Switch again
![image](https://user-images.githubusercontent.com/1897605/94653846-e1530900-0304-11eb-9ce5-6778466e92c1.png)
Now WSL write that network can't be find instead of 0xffffff error. Remove WSL Switch manually from Virtual Switch Manager. After next run WSL will create it by self.

Settings will be same as on screenshot but WSL can't work with it if you create it manually. Some hidden settings maybe

**Comment by th0ger** [comment]
> > I disable Hyper-v virtual Ethernet(VEthernet) in control panel, then start wsl2.
> > That error disappear and wsl start successfully.
> > then I enable virtual Ethernet, everything works fine for now.
> 
> This did it for me. ( I already uninstalled Docker before this step so not sure if uninstalling Docker is required as well).

This fix also works here, but not persisted on reboot.
I should note that the 0xffffffff error is caused by missing "vEthernet (WSL)" and WSL somehow is blocked from creating it.
When I disable "vEthernet (Default Switch)", WSL can create "vEthernet (WSL)".

**Comment by dcharlespyle** [comment]
I find it really odd that I have that problem each and every time after installing a new build, the last few builds, but that when I force the LxssManager service to stop from within Resource Monitor, and then restart the service again manually, I do not any longer have the problem at all, even after restarts, hibernation, or sleeping the machine and waking it again.  But then I don't use VPN or Docker, so that might possibly have something to do with it.  I still don't see the adapters in Network Connections but the connections still all are working.  And I just woke the computer from hibernation this morning.

![image](https://user-images.githubusercontent.com/5110398/95081610-05349700-06d7-11eb-9949-f92a0d3893d5.png)

![image](https://user-images.githubusercontent.com/5110398/95081802-53e23100-06d7-11eb-877e-c2667c86e8cb.png)

![image](https://user-images.githubusercontent.com/5110398/95081961-9277eb80-06d7-11eb-8fd1-6f182e810a31.png)

![image](https://user-images.githubusercontent.com/5110398/95081862-6ceae200-06d7-11eb-979a-f437a7c2ad68.png)

**Comment by barkermn01** [comment]
> After `docker network prune` run Hyper-V Manager -> Hyper-V Switch Manager -> Virtual Switches
> Create WSL Switch again
> Now WSL write that network can't be find instead of 0xffffff error. Remove WSL Switch manually from Virtual Switch Manager. After next run WSL will create it by self.
> 
> Settings will be same as on screenshot but WSL can't work with it if you create it manually. Some hidden settings maybe

This did not work for me, when i created it and tried running Ubuntu 20.04 i get the error: (somewhat expected)
> WslRegisterDistribution failed with error: 0x803b0001
> Error: 0x803b0001 The network was not found.

I remove it I get exactly the same as before
> WslRegisterDistribution failed with error: 0xffffffff
> Error: 0xffffffff (null)

Just updated to windows 10 2004

**Comment by QuAzI** [comment]
Windows lost WSL AGAIN. Just after reboot (updates was disabled from last issue).
Now WSL can't create right connection anymore. Even if I tried th0ger's solution
>  When I disable "vEthernet (Default Switch)", WSL can create "vEthernet (WSL)".

I saw in Network Connections that WSL tries to create new connection and removes it. But where is logs or something which can help to diagnose problems?
![20201013_093220_4OJtlsiau6](https://user-images.githubusercontent.com/1897605/95824267-ae166e00-0d37-11eb-974d-5e247820accd.gif)

**Comment by onomatopellan** [comment]
@QuAzI Take a look at Event Viewer -> Applications and Services -> Microsoft -> Windows -> Hyper-V-VmSwitch -> Operational

**Comment by QuAzI** [comment]
> 
> 
> @QuAzI Take a look at Event Viewer -> Applications and Services -> Microsoft -> Windows -> Hyper-V-VmSwitch -> Operational

Looks not very informative
[Hyper-V-VmSwith_Operational.zip](https://github.com/microsoft/WSL/files/5372509/Hyper-V-VmSwith_Operational.zip)

**Comment by onomatopellan** [comment]
@QuAzI It looks like the docker prune screwed everything. Open an elevated cmd and run `netcfg -d`. Reboot and you should have back the default network adapters.

**Comment by QuAzI** [comment]
@onomatopellan 
Thanks, `netcfg -d` works fine. Just two moments:
- VPN connections broken and must be recreated
- Device Manager -> Network adapters - All `Hyper-V Virtual Ethernet Adapter` should be removed before reboot

**Comment by Witti** [comment]
I removed Acrylic DNS Proxy from my system and wsl started working again.

**Comment by dcharlespyle** [comment]
I don't use any DNS program or VPN service and I still so far get the error after every upgrade to another build.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by mikegehard** [comment]
Uninstalled Acrylic DNS proxy and everything works as normal now. I guess it's manual hosts file entries for me.

**Comment by fjvela** [comment]
Same problem here, I resolve it stopping ICS service

**Comment by laurensiusadi** [comment]
Stopped Acrylic DNS service, and the WSL worked again

**Comment by yusufhm** [comment]
> I went through the same issue and it comes out it was caused by [Acrylic DNS Proxy](https://mayakron.altervista.org/support/acrylic/Home.htm).
> 
> Maybe services like that mess with networking settings necessary for Hyper-V to work (I noticed not even my plain Hyper-V VMs would come up).
> 
> I "solved" it by uninstalling Acrylic altogether and resorting to plain old hosts file editing.
> 
> Btw, I'm running Ubuntu 18.04.5 and Docker Desktop v19.03.12 on WSL2 on Windows 10 Build 19041.388.

Thanks for that; WSL stopped working for the same reason. What I ended up doing is set the service to start as delayed, which makes it start after a few minutes. That allows WSL to work again. If you need Acrylic earlier, you can always run Acrylic UI and start the service manually.

**Comment by iuriguilherme** [comment]
Everyone mentioning they "fixed the problem" in this issue didn't fix nothing at all.

In my case I had problems in the `~/.bash{rc,_profile}` files which prevented bash from starting and wsl didn't know what to do, thus presenting me this error code 4294967295. I had to use `wsl.exe -d Debian -e /bin/sh` to use another shell and fix the offending files.

So the problem here is that WSL2 distros lack automatic emergency shells and workarounds in the event of a problem in the system init (which it seems that inside Windows there are many more than regular Linux). Has anyone at @microsoft ever heard of busybox?

Regular Linux seem to have many layers of ways to prevent such failures, however when using Linux on Windows (where one would expect to see Linux crash much more frequently) there seems to be none.

**Comment by IncubusRK** [comment]
Check and stop any process listening 53/UDP 
https://github.com/microsoft/WSL/issues/4364#issuecomment-817124935

**Comment by ryanhelms** [comment]
For what it is worth, I noticed the Acrylic DNS Server situation above and I stopped my Technitium DNS Server and all of a suddden everything started working. 

Eeeeesh! 

Thanks for the help folks

**Comment by nashokin** [comment]
Tried every possible solution from this thread and many others, nothing worked for weeks.

I was receiving ' ... can't connect virtual machine container' error.

Fixed it by uninstalling 'Windows Subsystem for Linux Update' and then re-installing it through Windows Update(receive updates for other Microsoft products has to be enabled in Advanced options). Repairing didn't fix it, no need to restart. I'm on arm64 device, Surface Pro X.
![image](https://user-images.githubusercontent.com/6571738/128320822-da5d1f56-6aa8-4148-a450-55afe893009f.png)

**Comment by giovannicandido** [comment]
I have the same problem with Acrilic DNS. Have to setup its service to run manualy

**Comment by Ragarnoy** [comment]
Same issue, WSL broken, Docker broken as well as VirtualBox... and literally nothing has fixed it. Only option left is to do a complete reinstall of windows. This is extremely annoying.

**Comment by dcharlespyle** [comment]
I never had to reinstall the OS.  Have you tried removing the WSL connection using Hyper-V Virtual Switch Manager in Hyper-V Manager app, and stopping the LxssManager service using Resource Manager? Then reboot, and try again to run WSL.  It should have a new connection generated that may or may not show up in Hyper-V afterward, and should work.  A working configuration should look something like this:

![image](https://user-images.githubusercontent.com/5110398/141574702-0b215c42-9a13-4acc-9578-4a503910a0ec.png)

If a new connection doesn't show up, you can make a new one manually using the same as shown.

What was the last thing that you were doing before it broke? Update to new build of Windows 10 or upgrade to Windows 11? Some other update? If another update, try removing and reinstalling the update.  If you cannot reinstall, try reinstalling over the OS using the next update.  I haven't seen the problem since.

**Comment by Ragarnoy** [comment]
It honestly came out of nowhere, one day it worked, the next it didn't, it wasn't any major update that broke it.
I tried reinstalling the distro, and when I did the only I'd get from then on would be `WslRegisterDistribution failed with error: 0x8007000d`
I tried everything that was mentionned and more, I just reinstalled Hyper-V and WSL and it still wont let me install any new distro

**Comment by dcharlespyle** [comment]
Are you using an Insider Preview version of Windows? If so, if it expires WSL will generate this error.  It will appear out of the blue.  The other potential cause is a damaged file.  Check your hard drive for filesystem errors, including bad sectors.

Check the filesystem for corrupted files.  Check the system's image store for corruption.  Make sure that Windows Hypervisor Platform and Virtual Machine Platform are installed properly in addition to Hyper-V.  Also Containers.

![image](https://user-images.githubusercontent.com/5110398/141596445-f65ac779-4d96-4035-abb0-d05769336ee9.png)
![image](https://user-images.githubusercontent.com/5110398/141596522-3cce6c75-9ce0-48b8-a9e5-295edb204449.png)

Sometimes an update will cause the above three Optional Features not to be installed properly.  The only way to fix them, if they are the cause. is to disable them, reboot, and then enable them all again.

**Comment by jordantrizz** [comment]
> For what it is worth, I noticed the Acrylic DNS Server situation above and I stopped my Technitium DNS Server and all of a suddden everything started working.
> 
> Eeeeesh!
> 
> Thanks for the help folks

+1 for Stopping/Removing Technitium DNS Server. Conversion of a WSL 1 Ubuntu instance to WSL2 worked.

**Comment by svonjoi** [comment]
Solution to my particular case. windows 10 build 19043.1237
`wsl ~ -d debian` in my case is working. To solve this error I found that by default in windows terminal settings for debian, starting directory was set as `~` and I just changed it to windows format path, for example `C:\Users\User\Desktop` and this did the trick

**Comment by fedemarco** [comment]
My only solution so far was https://github.com/microsoft/WSL/issues/5092#issuecomment-743937383

I found that the tool detailed in that comment (sorry, not sure what is actually doing), finds and fixes a lot of errors in my git repositories.

At the same time, the git repositories are broken once I fix the disk.

It is clearly caused by that, but not sure the cause.

**Comment by baerla** [comment]
I had the same issue that Kali couldn't start.

My solution was:
I started Docker Desktop and was then able to login to Kali again

**Comment by tangruixun** [comment]
Thanks, it is indeed a DNS issue, after stopping Acrylic DNS service, not only WSL issue soloved, but also Windows Sandbox 0xfffffff problems disappears... 😅

**Comment by Leo137** [comment]
Yes, in my case i installed Technitium DNS Manager which operated in port 53. Which is a port apparently WSL needs to set up his network adapter.

Removing the program and restarting the system worked for me.

**Comment by ikamran** [comment]
I had the same issue it took lot of my time, simply check which app is using port 53 and kill it
 try the following:
Go to System Tools => Resource Monitor GUI for windows
Check which service is using port 53,
Once you identify it, kill the process as follows: taskkill /f /pid [PID].

Go to Users/[youruser]/AppData/Local/Packages/ and look for a folder called CanonicalGroupLimitedUbuntu... then right click on it, go to Properties => Advanced Options and disable compression for the folder, then click accept and also apply this change for subfolders.

Please try the following:

Go to System Tools => Resource Monitor GUI for windows
Check which service is using port 53,
Once you identify it, kill the process as follows: taskkill /f /pid [PID].
Go to Users/[youruser]/AppData/Local/Packages/ and look for a folder called CanonicalGroupLimitedUbuntu... then right click on it, go to Properties => Advanced Options and disable compression for the folder, then click accept and also apply this change for subfolders.
[Note] Windows uses Compression for the installation folders so is not possible to run it.

Once this is done, try to run your installation and should be working.

Hope it helps. Regards

**Comment by epenflow** [comment]
> I had the same issue it took lot of my time, simply check which app is using port 53 and kill it try the following: Go to System Tools => Resource Monitor GUI for windows Check which service is using port 53, Once you identify it, kill the process as follows: taskkill /f /pid [PID].
> 
> Go to Users/[youruser]/AppData/Local/Packages/ and look for a folder called CanonicalGroupLimitedUbuntu... then right click on it, go to Properties => Advanced Options and disable compression for the folder, then click accept and also apply this change for subfolders.
> 
> Please try the following:
> 
> Go to System Tools => Resource Monitor GUI for windows Check which service is using port 53, Once you identify it, kill the process as follows: taskkill /f /pid [PID]. Go to Users/[youruser]/AppData/Local/Packages/ and look for a folder called CanonicalGroupLimitedUbuntu... then right click on it, go to Properties => Advanced Options and disable compression for the folder, then click accept and also apply this change for subfolders. [Note] Windows uses Compression for the installation folders so is not possible to run it.
> 
> Once this is done, try to run your installation and should be working.
> 
> Hope it helps. Regards

ive same issue so far i can fix it with this, i think my problem is acrylicservice.exe that running on port 53. if you're using valet php on windows just type in cmd valet stop to kill acrlicservice.exe that running on port 53. WSL2 distros fail because of Acrylic DNS Proxy ([microsoft/wsl#4929](https://github.com/microsoft/WSL/issues/4929)). Use valet stop, start WSL2 then valet start.

https://packagist.org/packages/cretueusebiu/valet-windows

**Comment by nishantkyal** [comment]
> 

this worked for me. thanks

**Comment by nishantkyal** [comment]
> Once you identify it, kill the process as follows: taskkill /f /pid [PID].

this worked for me. thanks

**Comment by Kandeel4411** [comment]
Going to post my solution in case anyone had the same problem (This should also work as a guide for corrupted/broken filesystem); basically I installed a new GLIBC version in the WSL distro that resulted in it being non-compatible with WSL and  could no longer boot into the distro as it resulted in `Catastrophic Failure` error   and well, libc was broken :D!

Steps:
1.  First of all, create another WSL distribution (Add another new ubuntu for example) that will be used to help us mount the files to recover/debug

2.  Navigate to faulty distro `C:\Users\[user]\AppData\Local\Packages\[distro]\LocalState\[distroPackageName]` and copy the `vhdx` file there (it contains the broken filesystem) somewhere else like `C:\ext4.vhdx`

3. This step is optional but this will help you know if WSL is running and that if you need to shut it down or view errors
    - Go to `C:\Users\[user]`
    - Edit or create a file if not present called `.wslconfig` and add the following to the `[wsl2]` section:
    ```
    [wsl2]
    debugConsole=true
    ```
4. Mount the copied `vhdx` to windows
   - Open an admin powershell/cmd instance
   - Run `diskpart` then run the following
   ```
   select vdisk file="C:\ext4.vhdx"
   attach vdisk
   exit
   ```
   ![image](https://github.com/microsoft/WSL/assets/40201968/0b5c71fa-cb06-4278-833f-90ea980e0804)

   Cool now we can run some recovery tools or mount the system in order to perform edits/fixes.  In case of filesystem corruption, there is a magical tool called [TestDisk](https://www.cgsecurity.org/wiki/TestDisk_Download) which we will be using to check if there are any errors.
   - Download `TestDisk` then run it, you should see something like the following:

    ![image](https://github.com/microsoft/WSL/assets/40201968/f3e518c3-efd4-4acd-958b-b50f3965a10a)
    
    `\\.\PhsyicalDrive2` here was my attached vdisk, in your case it might be different
    - Press `Proceed` then `None`  and  you should see something like the following:
    
    ![image](https://github.com/microsoft/WSL/assets/40201968/dbf6f358-bcd3-4970-9d73-de2b27f5b6fb)
    You can press `List` and navigate to the files there in case you want to copy or check if its readable but in my case it was damaged (Even if it wasn't I suggest following the steps needed to perform repair by clicking `Superblock`)
    
    ![image](https://github.com/microsoft/WSL/assets/40201968/66959a7d-b440-43e2-98c0-b01115cb1ac2)
    
    As you can see, we need to run the `fsck.ext4 -p -b superblock -B blocksize device` command per superblock and to be able to run it, we need to mount it in a linux system (Thats where the 2nd new ubuntu distro comes in)
    
6.  Mount the vdisk into WSL by running  `wsl --mount \\.\PhysicalDrive2 --bare`  in an admin ps/cmd  (Might need to change vdisk name depending on your case)

7. Open your new linux distro and  download `TestDisk` on it so that we can differentiate which was our mounted vdisk.  Run the downloaded `TestDisk` by running in the testdisk directory `sudo ./testdisk_static`, you should see something like:
   
   ![image](https://github.com/microsoft/WSL/assets/40201968/da9a5c8c-9bcf-436a-849b-593869c73c31)
   
   - Now we can try to repair the filesystem, using the suggested `fsck` command per superblock, run the command and it should look something like:
   
    ```bash
    fsck -p -b 0, -B 4096 /dev/sdd
    fsck -p -b 32768, -B 4096 /dev/sdd
    fsck -p -b 98304, -B 4096 /dev/sdd
    fsck -p -b 163840, -B 4096 /dev/sdd
    fsck -p -b 229376, -B 4096 /dev/sdd
    fsck -p -b 294912, -B 4096 /dev/sdd
    fsck -p -b 819200, -B 4096 /dev/sdd
    fsck -p -b 884736, -B 4096 /dev/sdd
    fsck -p -b 1605632, -B 4096 /dev/sdd
    fsck -p -b 2654208, -B 4096 /dev/sdd
    ```
    You should run `TestDisk` again to try to list files and see if its able to, if it works then great! now we will actually mount the vdisk into WSL

8.  This step is optional, its for the case if you want to remove the offending files (In which case it was the custom installed `glibc` for me ). Unmount the mounted vdisk by running `wsl --unmount` and mount it again but this time without the `--bare` flag like `wsl --mount \\.\PhysicalDrive2` 
    - Open the new linux distro and cd to `/mnt/wsl` and you should see something like `PhsyicalDrive2` there, this is the mounted vdisk
    - In my case I had to go to `/mnt/wsl/PhsyicalDrive2/usr/local/lib` +`bin` + `sbin` and remove all the offending `libc` files there so that my system will use the default libc. Your cases may vary so do as you wish as you already have access to the files
    
9.  Final step, now we shall replace our old faulty `ext4.vhdx` with our newly modified one
     - Navigate to faulty distro `C:\Users\[user]\AppData\Local\Packages\[distro]\LocalState\[distroPackageName]` and copy the `vhdx` file there (it contains the broken filesystem) somewhere else for backup (IGNORE AT YOUR OWN RISK)
     - Copy the `ext4.vhdx`  that we fixed previously (e.g somewhere like `C:\ext4.vhdx`) and replace the one in the navigated distro with it
         -  If an access or in use error pops up, you need to first stop the faulty wsl instance if it was running by running `wsl --shutdown` and try again (If WSL continues to pop open then thats because of the annoying windows explorer triggering it, in that case, close the explorer and copy and replace using purely cmd/ps console)
         
     Run the faulty distro again and it should work as expected
     
10. Clean up
      - Remove `debugConsole=true` from `.wslconfig` if you ran this step
      - Detach vdisk from `diskpart`
         - Open an admin powershell/cmd instance
         - Run `diskpart` then run the following
         ```
         select vdisk file="C:\ext4.vhdx"
         detach vdisk
         exit
         ```
         
         
I hope this was able to help someone or helped be a reference on how to debug / recover from such an issue! Best regards

**Comment by kpnmn99** [comment]
I resolved the issue already.  I believe I removed and reinstalled WSL2 and that resolved the issue.  I was back up and running the next day or so.

Thanks for the responses.

KP
________________________________

**Comment by ByZain** [comment]
my solution is run the following command with administrator：
`netsh winsock reset`

**Comment by kpnmn99** [comment]
To whom it may concern,

I resolved this issue some time ago by terminating the VM and relaunching it.
In this case kali-linux

I ran this command from the PowerShell prompt.
wsl --terminate kali-linux

After relaunching, no issue.

Best regards.

I did reinstall WSL just in case.  But on a different I got the command, and just ran that PowerShell command.

**Comment by bolens** [comment]
I resolved this issue the following way:

1. I checked my wsl machines by running `wsl -l -v` and noticed that the `docker-desktop` machine was stuck uninstalling
```
  NAME                    STATE           VERSION
* Ubuntu                  Stopped         2
  docker-desktop          Uninstalling    2
  rancher-desktop         Stopped         2
  docker-desktop-data     Stopped         2
  kali-linux              Stopped         2
  podman-machine          Stopped         2
  rancher-desktop-data    Stopped         2
  Debian                  Stopped         2
```
2. I then ran the following to remove the stuck machine state `wsl --unregister docker-desktop`
```
Unregistering.
The operation completed successfully.
```
3. Ran Docker Desktop again which will recreate the machine

**Comment by kpnmn99** [comment]
Thanks for the response.  I resolved the issue as per a previous post.

Best regards.

KP
________________________________

**Comment by redhatted** [comment]
I then ran the following to remove the stuck machine state wsl --unregister docker-desktop

This worked for me after a day of trying different things! Thank you so much @bolens

**Comment by wasifferoze** [comment]
> The fix for me with this same issue was this:
> 
> 1. I checked my wsl machines by running `wsl -l -v` and noticed that the `docker-desktop` machine was stuck uninstalling
> 
> ```
>   NAME                    STATE           VERSION
> * Ubuntu                  Stopped         2
>   docker-desktop          Uninstalling    2
>   rancher-desktop         Stopped         2
>   docker-desktop-data     Stopped         2
>   kali-linux              Stopped         2
>   podman-machine          Stopped         2
>   rancher-desktop-data    Stopped         2
>   Debian                  Stopped         2
> ```
> 
> 2. I then ran the following to remove the stuck machine state `wsl --unregister docker-desktop`
> 
> ```
> Unregistering.
> The operation completed successfully.
> ```
> 
> 3. Ran Docker Desktop again which will recreate the machine

This works.

**Comment by OfficiallyCrazy** [comment]
> for me was solved like this: 1 - Uninstall docker; 2 - Uninstall / install Hyper-v; 3 - Uninstall / install subsystem;
> 
> started working again!

To me this works until i restart my pc 2-3 times. then it stops working again. everytime i lose my dockers

**Comment by vurso** [comment]
> > The fix for me with this same issue was this:
> > 
> > 1. I checked my wsl machines by running `wsl -l -v` and noticed that the `docker-desktop` machine was stuck uninstalling
> > 
> > ```
> >   NAME                    STATE           VERSION
> > * Ubuntu                  Stopped         2
> >   docker-desktop          Uninstalling    2
> >   rancher-desktop         Stopped         2
> >   docker-desktop-data     Stopped         2
> >   kali-linux              Stopped         2
> >   podman-machine          Stopped         2
> >   rancher-desktop-data    Stopped         2
> >   Debian                  Stopped         2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 2. I then ran the following to remove the stuck machine state `wsl --unregister docker-desktop`
> > 
> > ```
> > Unregistering.
> > The operation completed successfully.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 3. Ran Docker Desktop again which will recreate the machine
> 
> This works.

This worked for me on the latest Docker Desktop in Windows 11 I had to also unregister the `docker-desktop-data` image then restarted Docker Desktop which recreated those VM's and then made sure Docker Desktop is using Ubuntu as the backing distro.

**Comment by screenprog** [comment]
Step 1: `wsl.exe --install --no-distribution`
Step 2: reboot.

And it worked for me!
I hope it works for you as well.
